dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ExclusionFileParser.java

```diff
@@ -62,12 +68,29 @@
   static ImmutableList<LinkageErrorMatcher> parse(Path exclusionFile)
       throws SAXException, IOException {
 
+    File exclusion = exclusionFile.toFile();
+    validate(exclusion);
+
     XMLReader xmlReader = XMLReaderFactory.createXMLReader();
     ExclusionFileHandler handler = new ExclusionFileHandler();
     xmlReader.setContentHandler(handler);
 
-    InputSource inputSource = new InputSource(new FileInputStream(exclusionFile.toFile()));
+    InputSource inputSource = new InputSource(new FileInputStream(exclusion));
     xmlReader.parse(inputSource);
     return handler.getMatchers();
   }
+
+  private static void validate(File file) throws IOException, SAXException {
```

Comment:
This reads the file twice: once to validate and once to parse. Can this be combined into a single pass?

Reply:
I didn't think of a way to do that. ValidationDriver does not give a method related to XMLReader.

I thought reusing InputSource, but it's not intended to be reused:

>  so applications should not attempt to re-use such streams after they have been handed to a parser.

https://docs.oracle.com/javase/7/docs/api/org/xml/sax/InputSource.html#setPublicId(java.lang.String)